core/block.js

```diff
@@ -193,13 +222,13 @@ Blockly.Block.prototype.dispose = function(healStack) {
   }
   // Then dispose of myself.
   // Dispose of all inputs and their fields.
-  for (var i = 0, input; input = this.inputList[i]; i++) {
+  for (var j = 0, input; input = this.inputList[j]; j++) {
```

Comment:
I'm not a fan of this.  'j' makes me look for the enclosed loop.  At Google we have no problems with redeclaring a variable.  We prefer having lots of extra var statements since code can be edited without creating accidental globals.  From a performance perspective var is a no-op since variables are hoisted.  Other companies and linters have different opinions (some advocate that all vars must be at the top of the function), but the Google style is to use lots of var statements throughout the code.  See Closure's use:
https://github.com/google/closure-library/blob/master/closure/goog/array/array.js#L1581